app/services/frameworks/question.rb

```diff
@@ -0,0 +1,66 @@
+module Frameworks
+  class Question
+    attr_reader :source, :questions, :filepath
+
+    def initialize(filepath:, questions:)
+      @filepath = filepath
+      @source = YAML.safe_load(File.read(filepath))
+      @questions = questions
+    end
+
+    def key
+      @key ||= File.basename(filepath, '.yml')
+    end
+
+    def call
+      question.question_type = source['type']
+      build_options(source['options'] || [])
+
+      question.required = true if required?(source['validations'])
+      questions
+    end
+
+    def required?(validations)
+      validation_types = validations.flat_map { |validation| validation['type'] }
+      validation_types.any?('required')
+    end
+
+    def build_options(options)
+      options.each do |option|
+        question.options << option['value']
+
+        build_followup_question(
+          followups: option['followup'].presence || [],
+          value: option['value'],
+        )
+
+        build_followup_comments(
+          followup_comment: option['followup_comment'],
+          value: option['value'],
+        )
+      end
+    end
+
+    def build_followup_comments(followup_comment:, value:)
+      if followup_comment
+        question.followup_comment = true
+        question.followup_comment_options << value if required?(followup_comment.fetch('validations', []))
+      end
+    end
+
+    def build_followup_question(followups:, value:)
+      followups.each do |followup|
+        questions[followup] = questions[followup] || FrameworkQuestion.new(key: followup)
```

Comment:
we might not have built the question yet, so define the question if it doesn't exist or extract it if it does